base/promotion.jl

```diff
@@ -5,61 +5,66 @@
 """
     typejoin(T, S)
 
-Compute a type that contains both `T` and `S`.
+
+Return the closest common ancestor of `T` and `S`, i.e. the narrowest type from which
```

Comment:
I think this docstring could be made clearer. I'm trying to figure out exactly what `typejoin` and `promote_typejoin` are doing and I don't think the current descriptions are complete. The current version states that it returns

> the narrowest type from which they both [T and S] inherit

but that must always be `Union{T,S}`. However, my understanding is that `typejoin` doesn't return unions and that `promote_typejoin` was introduced exactly because of that. So what is this function returning? My guess would be something like

> `T` if `S<:T`
> `S` if `T<:S`
> else `min(TS)` for which `T<:TS` and `S<:TS` and `isabstract(TS)`

Reply:
I think I just copied "closest common ancestor" from a comment and tried to make it a little more explicit. You're right that `typejoin` returns an abstract supertype and never a `Union` AFAIK.